### PR TITLE
Bump rpm spec to beta7 release.

### PR DESCRIPTION
**Pull request by glima**
- move the namespace of the packages to soletta, not libsoletta (Fedora
   way dictates that)
- new sol-oic-gen.py bin added to devel package
- remove deprecated builtins.json
- add tinycbor dependency (submodule)
- add libmicrohttpd dependency (will trigger just on f23+)
- add http-{server,client} and json flow modules (server on f23+ only)
- don't ship config files anymore (instead sed the resulting .config)
- run checks too during the build

We got to figure out why between the _build_ and _install_ stages we regenerate the .h and such files again, it beats me (and it seems it's not `rpmbuild`'s fault).

Signed-off-by: Gustavo Lima Chaves gustavo.lima.chaves@intel.com

### CODE REVIEW

**Comment by glima** [comment]
pushed -f, since no comments here yet.

File: rpm/soletta.spec [hunk 1]
```diff
@@ -383,13 +412,14 @@ make CFLAGS="$CFLAGS %optflags" LDFLAGS="$LDFLAGS %__global_ldflags" %{?_smp_mfl
...
 
-%files -n lib%{name}-devel
+%files %{name}-devel
 %{_libdir}/libsoletta.so
 %{_bindir}/sol-fbp-generator
 %{_bindir}/sol-fbp-to-dot
 %{_bindir}/sol-flow-node-type-gen.py
-%{_bindir}/sol-flow-node-types
 %{_bindir}/sol-flow-node-type-validate.py
+%{_bindir}/sol-flow-node-types
```
Comment (anselmolsm) [inline rpm/soletta.spec:421]: I guess this binary should be removed from the install.

**Comment by anselmolsm** [comment]
LGTM, just sol-flow-node-types that shouldn't be installed (we should remove this from Soletta's repo).
And I am not a RPM specialist ;)

File: rpm/soletta.spec (hunk 1 above)
Comment (glima) [inline rpm/soletta.spec:421]: I need the commit in a tag 1st ;)

**Comment by glima** [comment]
Thanks :)

File: rpm/soletta.spec (hunk 1 above)
Comment (anselmolsm) [inline rpm/soletta.spec:421]: Ah, ok.

**Comment by glima** [comment]
Merged.

**Pull request closed without merging**